src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/ShipOutOfLuck.java

```diff
@@ -0,0 +1,156 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.SimpleHack;
+import com.programmerdan.minecraft.simpleadminhacks.configs.ShipOutOfLuckConfig;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.Block;
+import org.bukkit.command.CommandSender;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Vehicle;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.event.vehicle.VehicleMoveEvent;
+import org.bukkit.inventory.ItemStack;
+import vg.civcraft.mc.civmodcore.api.BlockAPI;
+import vg.civcraft.mc.civmodcore.api.ItemAPI;
+import vg.civcraft.mc.civmodcore.api.WorldAPI;
+
+public class ShipOutOfLuck extends SimpleHack<ShipOutOfLuckConfig> implements Listener {
+
+	public ShipOutOfLuck(SimpleAdminHacks plugin, ShipOutOfLuckConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public String status() {
+		StringBuilder builder = new StringBuilder(getClass().getSimpleName());
+		builder.append(" is ").append(isEnabled() ? "enabled" : "disabled").append(".").append("\n");
+		Set<Material> scuttleList = this.config.getScuttleList();
+		if (scuttleList.isEmpty()) {
+			builder.append("No scuttle blocks.");
+		}
+		else {
+			builder.append(" Scuttle blocks:\n");
+			builder.append(scuttleList.stream()
+					.map(effect -> "  • " + effect.name())
+					.collect(Collectors.joining("\n")));
+		}
+		return builder.toString();
+	}
+
+	// ------------------------------------------------------------
+	// Listeners
+	// ------------------------------------------------------------
+
+	@Override
+	public void registerListeners() {
+		this.plugin().registerListener(this);
+	}
+
+	@Override
+	public void unregisterListeners() {
+		HandlerList.unregisterAll(this);
+	}
+
+	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+	public void preventBoatPlacements(PlayerInteractEvent event) {
+		switch (event.getAction()) {
+			case RIGHT_CLICK_AIR:
+			case RIGHT_CLICK_BLOCK:
+				break;
+			default:
+				return;
+		}
+		Block placedOn = event.getClickedBlock();
+		ItemStack placed = event.getItem();
+		if (!BlockAPI.isValidBlock(placedOn) || !this.config.isScuttleBlock(placedOn.getType())
```

Comment:
You don't need to continue for RIGHT_CLICK_AIR if you're gonna require a valid block here

Reply:
Also once you've done that you should probably just remove these invalid checks entirely, they can never actually trigger. Nullcheck the item and then check if its a boat. If the block is not null it's never gonna be invalid either